src/nvim/msgpack_rpc/channel.c

```diff
@@ -823,6 +823,7 @@ static void log_server_msg(uint64_t channel_id,
   msgpack_unpack_next(&unpacked, packed->data, packed->size, NULL);
   uint64_t type = unpacked.data.via.array.ptr[0].via.u64;
   DLOGN("[msgpack-rpc] nvim -> client(%" PRIu64 ") ", channel_id);
+  log_lock();
```

Comment:
Why is channel.c managing `log.c`'s mutex? Can this go through `do_log` instead?